Pascal VOC generator image names reader fix

mxvs opened the pull request:
The current version of the PASCAL VOC generator does not process the format correctly, more specifically the image set files containing the image names are not handled properly.

Example lines from one of the image set files files:
```
2008_000003 -1
2008_000002 -1
2008_000003 -1
2008_000007 -1
2008_000008 -1
```

Only the first part is the actual image name that needs to be processed, running the current generator against the official PASCAL VOC dataset generates the following error(s):

```
FileNotFoundError: [Errno 2] No such file or directory: '/var/dl-data/VOCdevkit/VOC2012/JPEGImages/2008_000002 -1.jpg'
```

As you can see the second part of the line, the `-1` is also being read and considered as part of the image name.

This small patch fixes this issue by splitting the string and ensuring only the actual filenames are being parsed.

yhenon commented:
Thanks for the PR. Looking over the pascal voc dataset, it seems to be like this format:
```
2008_000003 -1
2008_000002 -1
2008_000003 -1
2008_000007 -1
2008_000008 -1
```
is only used for the class specific image sets, but not for the overall image sets. Is the intent of this PR the ability to train on just one of the classes by specifying for example `chair_train.txt` as your image set?

de-vri-es commented on keras_retinanet/preprocessing/pascal_voc.py:
```diff
@@ -81,7 +81,7 @@ def __init__(
         self.set_name             = set_name
         self.classes              = classes
-        self.image_names          = [l.strip() for l in open(os.path.join(data_dir, 'ImageSets', 'Main', set_name + '.txt')).readlines()]
+        self.image_names          = [l.strip().split(None, 1)[0] for l in open(os.path.join(data_dir, 'ImageSets', 'Main', set_name + '.txt')).readlines()]
```
I would prefer to write this as `str.split(maxsplit=1)` instead of `str.split(None, 1)`. I think the first version is much easier to understand without referring to the `str` documentation.

Also, what does the -1 even mean here? Why is it there?

hgaiser commented:
Any update on this ?

hgaiser commented:
I'll close this PR due to inactivity. It needs more feedback. Feel free to re-open if necessary.

mxvs commented:
Hello,

Sorry I was out of town and unable to reply. Its correct as mentioned by @yhenon that the original dataset containing ALL images does not have the additional character in the textfile. However when you want to train on a specific class subset it doesn't work with the current iterator.

More importantly when creating your OWN dataset using tools such as https://github.com/Microsoft/VoTT you also get the additional 1 or -1 and the generator does not currently work on files generated by this or other tools.

My change also does not break the existing code at all, it simply allows for the parsing of additional image sets in the original dataset or use datasets generated by tools like the one from Microsoft.

I also updated the change to use str.split(maxsplit=1) as indicated by @de-vri-es

de-vri-es commented:
I still wonder what this number means. Also, with this change files will not be able to have spaces in their names anymore. However, that seems to be a limit of the format itself, so there is little we can do about it here.

de-vri-es commented:
Thanks for the PR!
